code/game/objects/items/weapons/teleportation.dm

```diff
@@ -208,6 +187,40 @@ Frequency:
 		recharging = 1
 		processing_objects.Add(src)
 
+/obj/item/weapon/hand_tele/proc/choose_destination(var/user)
+	var/list/L = list(  )
+	for(var/obj/machinery/computer/teleporter/R in machines)
+		for(var/obj/machinery/teleport/hub/com in locate(R.x + 2, R.y, R.z))
+			if(R.locked && !R.one_time_use)
+				if(com.engaged)
+					L["[R.id] (Active)"] = R.locked
+				else
+					L["[R.id] (Inactive)"] = R.locked
+
+	var/list/turfs = new/list()
+
+	for (var/turf/T in trange(10, user))
+		// putting them at the edge is dumb
+		if (T.x > world.maxx - 8 || T.x < 8)
+			continue
+
+		if (T.y > world.maxy - 8 || T.y < 8)
+			continue
+
+		turfs += T
+
+	if (turfs.len)
+		L["None (Dangerous)"] = pick(turfs)
+
+	turfs = null
+	destination_name = input(user, "Please select a teleporter to lock in on.", "Hand Teleporter") in L
```

Comment:
I was wrong in an earlier comment: the sanity checks need to be moved after this `input` to prevent people from changing the destination after letting go of the hand tele.